samples/hybrid-connections/java/simple-http-demo/src/samples/HttpSender.java

```diff
@@ -0,0 +1,72 @@
+package samples;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.ExecutionException;
+
+import com.microsoft.azure.relay.StringUtil;
+import com.microsoft.azure.relay.TokenProvider;
+
+public class HttpSender {
+	static final String CONNECTION_STRING_ENV_VARIABLE_NAME = "RELAY_CONNECTION_STRING_ENVIRONMENT_VARIABLE";
+	static final Map<String, String> connectionParams = StringUtil.parseConnectionString(System.getenv(CONNECTION_STRING_ENV_VARIABLE_NAME));
+	static final String RELAY_NAME_SPACE = connectionParams.get("Endpoint");
+	static final String CONNECTION_STRING = connectionParams.get("EntityPath");
+	static final String KEY_NAME = connectionParams.get("SharedAccessKeyName");
+	static final String KEY = connectionParams.get("SharedAccessKey");
+	
+	public static void main(String[] args) throws IOException, InterruptedException, ExecutionException {
+		// Build a large test string over 64kb
+		StringBuilder builder = new StringBuilder();
+		String alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+		for (int i = 0; i < 2000; i++) {
```

Comment:
>2000 [](start = 22, length = 4)

Does builder have a way to determine its current size (not capacity)?
e.g. 
```
for (int i = 0; i < builder.getSize() <= 64 * 1024; i++) {
```

Reply:
I know this is less efficient code, but I think it's easier if the user wants to switch to say 2K or 100K message size.

---
In reply to: [242650792](https://github.com/Azure/azure-relay/pull/35#discussion_r242650792) [](ancestors = 242650792)